### PR TITLE
Pass optional `--file` argument to command docker.build

### PR DESCRIPTION
**Pull request by chribag**
I am probably missing something to publish.

### CODE REVIEW

File: src/noos_inv/docker.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
         "user": "AWS",
         "token": None,
         "name": "webserver",
+        "file": None,
```
Comment (pieeri) [inline src/noos_inv/docker.py:16]: Across the repo, only "sensitive" configs are left nullable (or `None`) and to be defined by OS environnement variables. Following this convention, the `file` config could be set at Docker's default `./Dockerfile`.

File: src/noos_inv/docker.py [hunk 2]
```diff
@@ -52,13 +53,16 @@ def _dockerhub_login(ctx: Context, user: str, token: Optional[str]) -> None:
...
 @task
-def build(ctx, name=None, context=None, arg=None):
+def build(ctx, name=None, file=None, context=None, arg=None):
     """Build Docker image locally."""
     name = name or ctx.docker.name
+    file = file or ctx.docker.file
     context = context or ctx.docker.context
     arg = arg or ctx.docker.arg
     utils.check_path(context)
     cmd = f"docker build --pull --tag {name} "
```
Comment (pieeri) [inline src/noos_inv/docker.py:63]: To add logic to check if the 'file' path exists in the file system (plus an additional small unit test?),

```python
utils.check_path(file)
```

File: src/noos_inv/docker.py [hunk 3]
```diff
@@ -52,13 +53,16 @@ def _dockerhub_login(ctx: Context, user: str, token: Optional[str]) -> None:
...
+    if file is not None:
```
Comment (pieeri) [inline src/noos_inv/docker.py:64]: ♻️ 

As discussed above in relation to a non-nullable value.

**Review by pieeri — CHANGES_REQUESTED** [review]
Small comment highlighting implicit convention (maybe to be  further documented) across the library for sensitive configs / secrets.

File: CHANGELOG.md [hunk 4]
```diff
@@ -62,3 +62,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.0.6] - 2021-01-05
 ### Changed
  - Bump Python version to v3.8.6.
+
+## [0.0.7] - 2021-05-20
```
Comment (pieeri) [inline CHANGELOG.md:66]: Version to bump accordingly in `noos_inv/__init__.py` and `pyproject.toml`.

File: src/noos_inv/docker.py [hunk 5]
```diff
@@ -52,13 +53,16 @@ def _dockerhub_login(ctx: Context, user: str, token: Optional[str]) -> None:
 
 
 @task
-def build(ctx, name=None, context=None, arg=None):
+def build(ctx, name=None, file=None, context=None, arg=None):
     """Build Docker image locally."""
     name = name or ctx.docker.name
+    file = file or ctx.docker.file
```
Comment (pieeri) [inline src/noos_inv/docker.py:59]: Maybe,

```python
file = file or f"{ctx.docker.context}/{ctx.docker.file}"
```

Comment (chribag) [reply src/noos_inv/docker.py:59]: isn't docker default PATH/Dockerfile and not dockercontext/Dockerfile ?

File: src/noos_inv/docker.py (hunk 2 above)
Comment (pieeri) [reply src/noos_inv/docker.py]: To get better clarity for distinguishing ill-formed function calls from failed underlying shell processes (not necessarily Python-based, thus uncertain about capturing correctly the stdout/stderr)?

**Review by pieeri — CHANGES_REQUESTED** [review]

**Review by pieeri — APPROVED** [review]

**Pull request merged**